wordpress-mobile/WordPress-iOS
# Limit Classic editor access on Simple WP.com sites to existing Classic posts

The next step after [notifying users that the classic editor is going away](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3209) is to actually remove it, starting with Simple sites.

I think this change doesn't warrant inclusion in RELEASE-NOTES.txt [since we already notified users about this change in 16.9](https://github.com/wordpress-mobile/WordPress-iOS/pull/16008), but if anyone thinks differently I'd love to hear. Perhaps we could include it in release notes but exclude it from the App Store version notes.

Addresses https://github.com/wordpress-mobile/gutenberg-mobile/issues/3092 (for iOS)

| <img src="https://user-images.githubusercontent.com/1898325/106808571-50d95c00-6649-11eb-9632-fcf3725db4c7.jpg" alt="The toggle switch" width="250"> |

This PR changes two things **on Simple sites only**:
1. Remove the "Use block editor" toggle switch from My Site → Settings on Simple sites.
2. Present the Gutenberg block editor when creating a new post, even on sites  which had defaulted to Classic in the app by turning the – now removed – "Use block editor" toggle off.

To reiterate, the above changes only affect Simple sites. They do not affect Atomic sites, Jetpack-connect self-hosted sites, regular self-hosted sites, etc. The classic editor will still be used in in the iOS share extension.

## To test

_Note: The toggle switch referred to below is the "Use block editor" toggle switch found in My Site → Settings. This is only available to site admins._

1. Using the app built from this branch, log into a WP.com account
2. Choose a Simple site where you are an admin
3. Verify that there **is no** toggle switch
4. Using the current App Store version of the app – not the app built from this branch – log into the same WP.com account
5. Choose the same Simple site where you are an admin, locate the toggle switch and turn it off (meaning use the classic editor) 
6. Moving back over to the app built from this branch – locate the same Simple site
7. Create a new post and ensure that the block editor is shown (not the classic editor)

## Regression notes (new ✨ )

I'm adding this section to try out @startuptester's proposal to improve code quality.

### Potential unintended areas of impact (regressions)
This PR removes the classic editor on Simple sites. However, we're leaving it in place for scenarios where the block editor cannot be used instead. These scenarios are a) editing a classic post (i.e. a post that contains only classic content), and b) creating a post via the iOS share extension. We have to check to make sure these scenarios still work on Simple sites. We should also check to make sure all other site types (Atomic/Jetpack/self-hosted) are unaffected by this change.

### What you did to manually test those/ what existing automated tests you relied on

I performed the following manual regression tests on a build made locally in Xcode from this branch. This is because one of the tests – the iOS sharing extension test – doesn't work on App Center builds (I get a "Sharing error" and I don't think it's related to the changes I've made in this PR, I've seen it before).

#### Regression test Simple sites
1. Log into a WP.com account and choose a Simple site
2. Locate an existing post that contains _only_ classic content
3. Open this post for editing and ensure that the classic editor is opened
4. Background the app and open the Photos app
5. Select a photo and share it to WordPress
6. Verify that the classic editor opens as expected

####  Regression test Atomic sites
Our UI tests will ensure that Atomic sites still show a toggle switch and that the switch works, i.e. it controls the default editor on mobile (feel free to run the `EditorAztecTests` and `EditorGutenbergTests` UI tests locally in Xcode to check this). All that's left is to check that existing posts that contain classic content are opened in the classic editor:

1. Log into a WP.com account and choose an Atomic site
2. Locate an existing post that contains _only_ classic content
3. Open this post for editing and ensure that the classic editor is opened

#### Regression test Jetpack-connected self-hosted sites
1. Log into a WP.com account in the app
2. Choose a Jetpack-connected self-hosted site and verify
    - that there **is** a toggle switch
    - that the toggle switch still controls which editor is used for new posts
    - that existing posts that contain classic content are opened in the classic editor

#### Regression test self-hosted sites
1. Log into a self-hosted site in the app where you are an admin (ensure you are using the "Enter your existing site address" option to log in)
2. Verify
    - that there **is** a toggle switch
    - that the toggle switch still controls which editor is used for new posts
    - that existing posts that contain classic content are opened in the classic editor

### What automated tests you added or what prevented you from doing so

Our current UI test suite includes tests for both the classic editor (a.k.a. Aztec) and the block editor (a.k.a Gutenberg). These tests run on just one site type: Simple sites. Now that classic is no longer an option for new posts on Simple sites, ~I updated the all UI tests to run on an Atomic site instead. This is was the easiest option to implement, but changes our entire UI test suite, so a less risky option could be to just update the classic editor tests to run on n Atomic site and leave the other UI tests to run on a Simple site as they currently do.~ We are disabling the Classic Editor tests until they can be migrated to executing on existing classic posts, instead of creating new ones. That work is tracked with this ticket here: https://github.com/wordpress-mobile/WordPress-iOS/issues/16218

## PR submission checklist
